test/support/fixtures/o_auth/o_auth_fixtures.ex

```diff
@@ -77,11 +77,16 @@ defmodule Teiserver.OAuthFixtures do
   given a user id, will create a OAuth application and a valid token for that
   user, returning both
   """
-  def setup_token(%{user: user}), do: setup_token(user.id)
-  def setup_token(%{id: id}), do: setup_token(id)
+  def setup_token(user_or_id, opts \\ [])
+  def setup_token(%{user: user}, opts), do: setup_token(user.id, opts)
+  def setup_token(%{id: id}, opts), do: setup_token(id, opts)
```

Comment:
When would this `def setup_token(%{id: id}, opts), do: setup_token(id, opts)` get called? What map other than user would have the id key for this to match?

Reply:
iirc, this is to allow [some setup code](https://github.com/geekingfrog/teiserver/blob/fbf05891476945a759e84b41573037819a826ae4/test/teiserver_web/controllers/tachyon_controller_test.exs#L144) with an access to a user to directly call it. It's more for convenience, so you can pass a user (map with an id) or just the id.

This change adds the option to specify the scopes for the token.